### PR TITLE
Move runtime sharding into config variable

### LINKED ISSUE #75: Turn on runtime sharding by default
1. This should probably be on by default

2. It's not clear if it should really be an argument to the executor or in the config or what.

### PR DESCRIPTION
**Pull request by shivaram**
This closes #75

### CODE REVIEW

**Comment by shivaram** [comment]
Couple of things that came to my mind while making this change

- While we generate the default config, should we set `num_shards` to 50 ?
- We need to document what all the valid config variables are and where they are used. I will do this as a part of #86

**Comment by ooq** [comment]
I actually meant that `num_shards` should completely be in `[runtime_name].meta.json` that is stored in S3 and the client code should be completely hidden from sharding. 
This would also make switching between runtimes easier.

**Comment by shivaram** [comment]
Ah I see. Ok - @ericmjonas Is that what you had in mind as well ? I can make that change but I think we need to be careful then to make sure runtimes generated before pywren 0.2 are still compatible etc.

**Comment by ericmjonas** [comment]
Yes that's what i was thinking -- do we care much about previous-generation
runtimes? can we default try and get the key in the json and fallback
gracefully to nonsharding?

**Comment by shivaram** [comment]
Yep - I will try to do that. It'll require a change to the runtimes repo, but it should be doable.

**Comment by shivaram** [comment]
@ericmjonas @ooq I updated this, but this depends on https://github.com/pywren/runtimes/pull/2 -- So I'd suggest reviewing that first. 

This is backwards compatible in that if the metadata doesn't have a `urls` field we just fall back to the base runtime url. The one catch though is that all the lambdas will need to access the metadata. I considered shipping that as a part of the job description, but I was worried if it would be too big for the 128KB limit ?

**Comment by ericmjonas** [comment]
@shivaram I'm a little confused here why the lambdas need access to the metadata, can't we just pass them a runtime key as part of the invoker data?

**Comment by ericmjonas** [comment]
If we have the executor, upon creation, read the `meta.json` from s3, then we will: 
1. handle serialization better in the future (As we will know at the executor level which packages are remote installed)
2. be able to just pass the lambdas a straight-up s3 URL and not have them worry about randomly selecting which one to use, and thus they will never have to hit `meta.json`

**Comment by shivaram** [comment]
@ericmjonas Your comment totally makes sense. Updated now

**Comment by shivaram** [comment]
Also I'll add the module check in a separate PR. I still need to understand the cloudpickle code a bit more

File: pywren/wren.py [hunk 1]
```diff
@@ -68,7 +70,9 @@ def remote_executor(config= None, job_max_runtime=3600,
     S3_PREFIX = config['s3']['pywren_prefix']
     invoker = invokers.SQSInvoker(AWS_REGION, SQS_QUEUE)
     return Executor(AWS_REGION, S3_BUCKET, S3_PREFIX, invoker, config,
-                    job_max_runtime, shard_runtime=shard_runtime)
+                    job_max_runtime)
+
+standalone_executor = remote_executor
```
Comment (shivaram) [inline pywren/wren.py]: is this line necessary ? just checking as it wasn't in the original PR if i'm right

File: pywren/wren.py [hunk 2]
```diff
@@ -6,6 +6,9 @@
     import pickle
 from tblib import pickling_support
 import logging
+import botocore
```
Comment (shivaram) [inline pywren/wren.py]: are these imports related to this PR ?

File: pywren/wren.py (hunk 1 above)
Comment (ooq) [inline pywren/wren.py]: I think I removed it during refactoring.
Piggybacking on your PR to add it back...(bad practice)

File: pywren/wren.py (hunk 2 above)
Comment (ooq) [inline pywren/wren.py]: No. Don't know how it shows up. Now almost every file in the project has unused imports. I'm planning to get rid off them in the PyLint PR, which is coming...very soon.
